grpcproxy/grpcproxy.go

```diff
@@ -0,0 +1,106 @@
+package grpcproxy
+
+import (
+	"github.com/grpc-ecosystem/grpc-gateway/runtime"
+
+	"fmt"
+	"plugin"
+	"strings"
+
+	"github.com/akutz/gpds/lib"
+	"golang.org/x/net/context"
+	"google.golang.org/grpc"
+)
+
+func LoadGRPCProxyPlugin(path string, targetURL string, gRPCProxyMux *runtime.ServeMux) error {
+	// Load our plugin
+	p, err := plugin.Open(path)
+	if err != nil {
+		return err
+	}
+
+	// lookup the plug-in's Types symbol; it's the type map used to
+	// register the plug-in's modules
+	tmapObj, err := p.Lookup("Types")
+	if err != nil {
+		return fmt.Errorf("error: failed to lookup type map: %v", err)
+	}
+
+	// assert that the Types symbol is a *map[string]func() interface{}
+	tmapPtr, tmapOk := tmapObj.(*map[string]func() interface{})
+	if !tmapOk {
+		return fmt.Errorf("error: invalid type map: %T", tmapObj)
+	}
+
+	// assert that the type map pointer is not nil
+	if tmapPtr == nil {
+		return fmt.Errorf("error: nil type map: type=%[1]T val=%[1]v", tmapPtr)
+	}
+
+	// dereference the type map pointer
+	tmap := *tmapPtr
+
+	// register the plug-in's modules
+	for k, v := range tmap {
+		lib.RegisterModule(k, v)
+	}
+
+	// Instantiate a copy of the module registered by the plug-in.
+	modGo := lib.NewModule("register")
+
+	ctx := context.Background()
```

Comment:
Related to the question below on the `Get/Set` interface - why is a context in the mix? We don't seem to make any use of it. And it seems unnecessary for getter/setters.

Reply:
I really don't want to mess with these because it's part of the plugin fix for type passing - it's literally a lift and shift from the `github.com/akutz/gpds/lib` code.

Reply:
I've had a look and it seems like context isn't used there at all. Are we going to make use of it? If not, I'd leave it out of the API.